space/authz/authz.go

```diff
@@ -18,97 +22,108 @@ import (
+func NewAuthzService(config auth.ServiceConfiguration) *KeycloakAuthzService {
+	return &KeycloakAuthzService{config: config, doer: rest.DefaultHttpDoer()}
 }
 
 // Configuration returns authz service configuration
-func (s *KeycloakAuthzService) Configuration() AuthzConfiguration {
+func (s *KeycloakAuthzService) Configuration() auth.ServiceConfiguration {
 	return s.config
 }
 
-// Authorize returns true and the corresponding Requesting Party Token if the current user is among the space collaborators
-func (s *KeycloakAuthzService) Authorize(ctx context.Context, entitlementEndpoint string, spaceID string) (bool, error) {
+// Authorize returns true if the current user is among the space collaborators
+func (s *KeycloakAuthzService) Authorize(ctx context.Context, spaceID string) (bool, error) {
 	jwttoken := goajwt.ContextJWT(ctx)
 	if jwttoken == nil {
 		return false, errors.NewUnauthorizedError("missing token")
 	}
-	return s.checkEntitlementForSpace(ctx, *jwttoken, entitlementEndpoint, spaceID)
+	return s.checkRole(ctx, *jwttoken, spaceID)
 }
 
-func (s *KeycloakAuthzService) checkEntitlementForSpace(ctx context.Context, token jwt.Token, entitlementEndpoint string, spaceID string) (bool, error) {
+type Roles struct {
+	Data []Role `json:"data"`
+}
+
+type Role struct {
+	RoleName   string `json:"role_name"`
+	AssigneeID string `json:"assignee_id"`
+}
+
+func (s *KeycloakAuthzService) checkRole(ctx context.Context, token jwt.Token, spaceID string) (bool, error) {
 	if !s.config.IsAuthorizationEnabled() {
 		// Keycloak authorization is disabled by default in Developer Mode
 		log.Warn(ctx, map[string]interface{}{
 			"space_id": spaceID,
 		}, "Authorization is disabled. All users are allowed to operate the space")
 		return true, nil
 	}
-	resource := auth.EntitlementResource{
-		Permissions: []auth.ResourceSet{{Name: spaceID}},
+	currentIdentityID, err := login.ContextIdentity(ctx)
+	if err != nil {
+		return false, err
 	}
-	ent, err := auth.GetEntitlement(ctx, entitlementEndpoint, &resource, token.Raw)
+
+	req, err := http.NewRequest("GET", fmt.Sprintf("%s/resources/%s/roles", s.config.GetAuthServiceURL(), spaceID), nil)
 	if err != nil {
 		return false, err
 	}
-	return ent != nil, nil
+	req.Header.Set("Authorization", "Bearer "+token.Raw)
+	res, err := s.doer.Do(ctx, req)
+	if err != nil {
+		return false, errors.NewInternalError(ctx, err)
+	}
+	defer rest.CloseResponse(res)
+	bodyString := rest.ReadBody(res.Body)
+	if res.StatusCode != http.StatusOK {
+		return false, errors.NewInternalError(ctx, errs.New("unable to get space roles. Response status: "+res.Status+". Response body: "+bodyString))
+	}
+
+	var roles Roles
+	err = json.Unmarshal([]byte(bodyString), &roles)
+	if err != nil {
+		return false, errors.NewInternalError(ctx, err)
+	}
+
+	id := currentIdentityID.String()
+	for _, r := range roles.Data {
+		if r.AssigneeID == id && (r.RoleName == "admin" || r.RoleName == "contributor") {
```

Comment:
Would be nice if we could take the business logic out of this method so that we can re-use this method.

We could do something like : 
```
func hasRole(rolename string){
        allRoles = getRoles(..)
        if rolename in allRoles {
                 return true
        }
}
```

Reply:
Well. This code is temporal anyway. We won't check roles when we switch to the new AuthZ flow. And we don't need to re-use this (hasRole) code for now. I would keep it simple until we have a real use case.